### ISSUE TITLE
Laggy text-editing for larger notes

### ISSUE THREAD

**Issue by mleue** [notable/notable#232]
First of all, thanks for this great program and for sharing it with us.

I have noticed that for larger files, text-editing can become quite laggy.
E.g. if you generate 300 paragraphs of lorem ipsum here: https://getlorem.com/ (resulting in 600 lines of text with ~25k words in total) and paste it into a new file in notable, text-editing noticeably slows down for me in that file.

For me this arrived naturally in some files that basically keep all my notes on some technical book that I've been reading.

Now maybe I'm just using notable the wrong way and I should just turn those huge files into a tag and then add every note as a separate file under that tag, but that's not really how I would like to organize my thoughts.

So basically I guess I want to ask if you/anyone has observed something similar (i.e. make sure it's not just my system) and if so, if there might be some way forward to making notable faster even for large notes. (maybe I'm just spoiled by vim?)

I'm on Ubuntu 18.04 running on a 2018 Dell XPS 13 with an i7, SSD, 16GB RAM ... never had any speed issues when editing text before. :)

Thanks in advance.

**Comment by fabiospampinato** [comment]
This is definitely something that shouldn't happen. There might be multiple reasons behind, maybe we are rendering too many lines at once, maybe CodeMirror is not fast enough, or something else.

**Comment by RomanKornev** [comment]
Can confirm on Windows 10 x64 1809 with i7 8700K 32GB RAM.
Editing long text file and holding any letter down maxes out the CPU core.
As a result, Notable slows down to a crawl and takes a while to 'unhang'.

**Comment by bendem** [comment]
In case anyone is interested, here is a profile trace that you can load in devtools.

https://gist.github.com/bendem/86dc2463a36aaa064f0a9c41758887dc

**Comment by bendem** [comment]
If I do that, I can't edit a file that's larger than 120KB without the application hanging. Also the application gets in a pretty broken state (scrollbar is wrong or missing, ghost text appears when scrolling, editing viewport is larger than its container).

**Comment by fabiospampinato** [comment]
@bendem there might be some CSS conflicting with CodeMirror. I'm not sure it's worth looking too much into this issue, maybe we should rather accelerate the switch to Monaco #203

**Comment by fantomc0der** [comment]
Just came across this issue today (on Win10 x64 i7-4790 32GB RAM) when moving an existing markdown file without much formatting (~900 lines) into Notable. After pasting the note, I tried changing the title and realized that keystrokes were taking so long to appear that it was basically unusable. Seems like there's a consistent keystroke lag that correlates to the size of the note after trying to remove a few chunks of lines at a time.

**Comment by Spidersouris** [comment]
I use Notable on a daily basis (Windows 10, i5-4570, GTX 970, 16 GB RAM) and write hundred of words each day. Right now, the main file I use has approximately 3000 words and it is considerably slow when typing. Creating a new blank file obviously fixes the issue for that file, but as soon as I copy/paste what I've written in my main file, it starts to lag, so it seems to be related to the size of the file.

**Comment by nlachmann** [comment]
Really nice tool! I'd love to use it more, but at the moment I can only create rather small notes with it.

Once the text grows beyond 1000-2000 chars, typing is laggy (like 500ms). If the note has more than 20k chars typing like 5 chars in a row chars lags 5-6 seconds behind before they are displayed all at once  the editor. Scrolling and displaying is fast, however. I **don't** see a huge CPU spike during typing, though. CPU goes up like 5-10% only.

**Comment by fabiospampinato** [comment]
I'm in the process of switching to Monaco #203, and I've just edited a 12MB note with 200k+ lines in it, the editor just doesn't break a sweat it seems. There are still some performance optimizations to be had on this front, for instance I think currently some event handlers are called too frequently, but I'd say the difference is pretty stark already.

**Comment by fabiospampinato** [comment]
We just switched to Monaco and this is no longer a problem.
